### ISSUE TITLE
Meta: Edge Anti-Aliasing Issues (AA1)

### ISSUE THREAD

**Issue by RedDevilus** [PCSX2/pcsx2#5069]
### Describe the Bug

Test

### Reproduction Steps

to

### Expected Behavior

be replaced

### PCSX2 Revision

a

### Operating System

Windows 10 (32bit)

### If Linux - Specify Distro

a

### CPU

a

### GPU

a

**Comment by gilderoylockhart01** [comment]
Metal Gear Solid 2 (3?) and OSDSYS are also valid tests for that meta. I guess they now should be fine, I will test that later.

**Comment by RedDevilus** [comment]
Doing software mode and switching edge anti-aliasing option is a good way to detect if it's that.

**Comment by gilderoylockhart01** [comment]
That's the case for MGS2. But I would like if someone else that knows the game better can confirm.

**Comment by gilderoylockhart01** [comment]
Ok so, not every games are fine with the hardware render yet, FFX and MGS 2 are still having minor problems with AA1 which is not the case with the software render.

**Comment by refractionpcsx2** [comment]
yes they won't be, because they do AA1 on triangles, something we can't emulate easily. only games which used AA1 lines are fine now.

**Comment by TransparentBlue** [comment]
Parts of the HUD in Ace Combat 4 is either the wrong color (too dark) or blinks a lot depending on whether edge anti-aliasing is turned on in SW, neither being  correct, HW mode is correct however.

**Comment by gilderoylockhart01** [comment]
One title "Trapt", makes abusive use of Edge AA on triangles, to the point where entire areas of the 3d scene are missing on the hardware render (and obviously software without edge AA). Surprising nobody reported it before.

**Comment by RedDevilus** [comment]
https://cdn.knightlab.com/libs/juxtapose/latest/embed/index.html?uid=eb0d1130-0113-11ec-abb7-b9a7ff2ee17c

There is a juxtapose of trapt but it is known.

**Comment by gilderoylockhart01** [comment]
Sure, actually this bug happens a lot in the introduction scene:

Hardware:
![gs_20211218180621](https://user-images.githubusercontent.com/90222976/146649635-dfd71032-0087-434b-a03a-bfa80cdc8615.png)

Software without edge AA:
![Capture d’écran 2021-12-18 180735](https://user-images.githubusercontent.com/90222976/146649642-a5678ebe-f53e-4895-b551-cb650bf01492.png)

Software with edge AA:
![2](https://user-images.githubusercontent.com/90222976/146649648-2122a53a-6f82-45be-b71f-66316beac6a9.png)

**Comment by gilderoylockhart01** [comment]
Metal Gear Solid 2 weapon laser is fixed with edge AA on textures with the hardware render.

**Comment by gilderoylockhart01** [comment]
And we can add one more entry, and that was a tricky one, the BIOS itself.

In the save menu, there are very rare cases where edge on triangles is used, take spider friend or foe and it's save icon, look how the edges are renders with edge aa:

SW without edge aa:
![sw no aa](https://user-images.githubusercontent.com/90222976/150656255-a097f760-74b0-4e4c-8dba-d18b4ed7075f.png)

SW with edge aa:
![sw aa](https://user-images.githubusercontent.com/90222976/150656260-b099575d-d208-4a8e-b125-cb45323b8858.png)

HW:
![hw](https://user-images.githubusercontent.com/90222976/150656262-24abb273-8542-4f93-86df-a12af279a9fc.png)

**Comment by xiriby** [comment]
Hey Can you please add Virtua Quest to this list

https://github.com/PCSX2/pcsx2/issues/5721

**Comment by JordanTheToaster** [comment]
GT3 Aspect NTSC-J HUD AA1.

![image](https://github.com/PCSX2/pcsx2/assets/80843560/5e468e6f-5b03-4894-8b36-223439afd2cc)

**Comment by SlyCooperReloadCoded** [comment]
Jonny Moseley Mad Trix and Portal Runner use anti-alias pixels to form their debug wireframe view. PR #6636 fixes this in software, however it has yet to be implemented in hardware, so the draw is just skipped.

**Comment by wxvu** [comment]
Armored Core series uses AA1 for the following:
- Lock-on box (All titles)
- Radar boundary lines (All titles)
- Pause screen (Armored Core 3 & Silent Line)
- Message box (Armored Core 2 & Another Age)

See: https://github.com/PCSX2/pcsx2/pull/13681#issuecomment-3644397553
[Armored Core 2_SLUS-20014_20251211181859.gs.zip](https://github.com/user-attachments/files/24117402/Armored.Core.2_SLUS-20014_20251211181859.gs.zip)

**Comment by wxvu** [comment]
Steambot Chronicles uses AA1 for general rendering.
Fixed by #13681.

[Steambot Chronicles_SLUS-21344_AA1boat.gs.zip](https://github.com/user-attachments/files/24784854/Steambot.Chronicles_SLUS-21344_AA1boat.gs.zip)

**Comment by AgentLogan1** [comment]
<img width="552" height="105" alt="Image" src="https://github.com/user-attachments/assets/eabf247a-a972-4302-ade7-09951da6fb72" />

> Doing software mode and switching edge anti-aliasing option is a good way to detect if it's that.

May I ask why some of the listed games at the top have a check mark but some have an empty box next them?
Also I can't find the "Edge Anti-Aliasing" option to test it.
I can help with testing Syphon Filter: Dark Mirror, since there is a PR open now to fix this issue for the listed games, just don't know the reproduction steps, what should I be looking for?

Edit: Noting Dark Mirror PAL and Logan's Shadow might share the issue as well.

**Comment by RedDevilus** [comment]
> <img width="552" height="105" alt="Image" src="https://github.com/user-attachments/assets/eabf247a-a972-4302-ade7-09951da6fb72" />
> 
> > Doing software mode and switching edge anti-aliasing option is a good way to detect if it's that.
> 
> May I ask why some of the listed games at the top have a check mark but some have an empty box next them?
> Also I can't find the "Edge Anti-Aliasing" option to test it.
> I can help with testing Syphon Filter: Dark Mirror, since there is a PR open now to fix this issue for the listed games, just don't know the reproduction steps, what should I be looking for?
> 
> Edit: Noting Dark Mirror PAL and Logan's Shadow might share the issue as well.

It's hardware-level feature that smooths the edges of primitives (lines and triangles also by extension the strips). 


The meta doesn't auto update the checkboxes first of all and the Edge Anti-aliasing option was merged to always be enabled. There is also no real benefit to let it be toggled for performance or compatibility reasons. It's easiest to see in the BIOS with the memcard icons, as it can differ per game (ex. FFX has both)

Software mode can do everything since a prior PR and was expanded partially to the hardware modes so it isn't fixed entirely.

If you have other examples for other games it's best to post a GS dump though not sure if it will help alot for a meta issue as this one.

**Comment by JordanTheToaster** [comment]
Hardware now has AA1 support so most of these should be working with that enabled.

**Comment by MarioSonic2987** [comment]
PS2 BIOS comparison

<details>
 <summary>Images</summary>

**AA1 disabled:**
<img width="852" height="731" alt="Image" src="https://github.com/user-attachments/assets/81ab40cb-0614-4ec9-b8de-b5d7ebfe899c" />

**AA1 enabled:**
<img width="852" height="731" alt="Image" src="https://github.com/user-attachments/assets/d3270836-b253-417e-a912-e3b810716245" />

**Software:**
<img width="852" height="731" alt="Image" src="https://github.com/user-attachments/assets/d262d80a-73bc-404d-a2ec-2aa9db3a24dc" />
</details>
